components/VercelChat/ChatInput.tsx

```diff
@@ -22,44 +26,61 @@ export function ChatInput({
+  const { hasPendingUploads } = useVercelChatContext();
   const isDisabled = !selectedArtist;
 
-  const handleSend = () => {
-    if (input === "" || isDisabled) return;
+  // Create a form ref to submit the form programmatically
+  const formRef = useRef<HTMLFormElement>(null);
+
+  const handleSend = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (input === "" || isDisabled || hasPendingUploads) return;
 
     if (isGeneratingResponse) {
       onStop();
     } else {
-      onSendMessage();
+      onSendMessage(event);
     }
   };
 
   return (
-    <div className="w-full relative p-3 dark:bg-zinc-800 rounded-2xl flex flex-col gap-1 bg-zinc-100">
-      <Input
-        input={input}
-        setInput={setInput}
-        isGeneratingResponse={isGeneratingResponse}
-        onSend={handleSend}
-        isDisabled={isDisabled}
-      />
-
-      <div className="absolute bottom-2.5 right-2.5">
-        <button
-          type="button"
-          className={cn(
-            "size-8 flex flex-row justify-center items-center dark:bg-zinc-100 bg-zinc-900 dark:text-zinc-900 text-zinc-100 p-1.5 rounded-full hover:bg-zinc-800 dark:hover:bg-zinc-300 hover:scale-105 active:scale-95 transition-all",
-            {
-              "dark:bg-zinc-200 dark:text-zinc-500 cursor-not-allowed opacity-50":
-                isGeneratingResponse || input === "" || isDisabled,
-            }
-          )}
-          onClick={handleSend}
-          disabled={isGeneratingResponse || input === "" || isDisabled}
-        >
-          {isGeneratingResponse ? <StopIcon /> : <ArrowUpIcon />}
-        </button>
+    <div>
+      <div className="w-full">
+        <AttachmentsPreview />
       </div>
+      <form
+        ref={formRef}
+        className="w-full relative p-3 dark:bg-zinc-800 rounded-2xl flex flex-col gap-1 bg-zinc-100"
+        onSubmit={handleSend}
+      >
+        <div className="absolute bottom-2.5 left-2.5 z-10">
+          <PureAttachmentsButton />
+        </div>
+        <Input
+          input={input}
+          setInput={setInput}
+          isGeneratingResponse={isGeneratingResponse}
+          onSend={() => formRef.current?.requestSubmit()}
+          isDisabled={isDisabled || hasPendingUploads}
+        />
+
+        <div className="absolute bottom-2.5 right-2.5">
+          <button
+            type="submit"
+            className={cn(
+              "size-8 flex flex-row justify-center items-center dark:bg-zinc-100 bg-zinc-900 dark:text-zinc-900 text-zinc-100 p-1.5 rounded-full hover:bg-zinc-800 dark:hover:bg-zinc-300 hover:scale-105 active:scale-95 transition-all",
+              {
+                "dark:bg-zinc-200 dark:text-zinc-500 cursor-not-allowed opacity-50":
+                  isGeneratingResponse || input === "" || isDisabled || hasPendingUploads,
+              }
+            )}
+            disabled={isGeneratingResponse || input === "" || isDisabled || hasPendingUploads}
```

Comment:
this logic is repeated 2x 
- please follow the DRY principle by creating a shared variable which is easier to maintain.